great_expectations/zep/postgres_datasource.py

```diff
@@ -65,64 +84,158 @@ def get_batch_request(
-                Leaving this empty, "", will add the options to the global namespace.
+            default_year_range: When this splitter is used, say in a BatchRequest, if no value for
+                year is specified, we query over all years in this range.
+                will query over all the years in this default range.
+            default_month_range: When this splitter is used, say in a BatchRequest, if no value for
+                month is specified, we query over all months in this range.
 
         Returns:
             This TableAsset so we can use this method fluently.
         """
         self.column_splitter = ColumnSplitter(
             method_name="split_on_year_and_month",
             column_name=column_name,
-            name=name,
-            template_params=["year", "month"],
+            param_defaults={"year": default_year_range, "month": default_month_range},
         )
         return self
 
+    def fully_specified_batch_requests(self, batch_request) -> List[BatchRequest]:
+        """Populates a batch requests unspecified params producing a list of batch requests
+
+        This method does NOT validate the batch_request. If necessary call
+        TableAsset.validate_batch_request before calling this method.
+        """
+        if self.column_splitter is None:
+            # Currently batch_request.options is complete determined by the presence of a
+            # column splitter. If column_splitter is None, then there are no specifiable options
+            # so we return early.
+            # In the future, if there are options that are not determined by the column splitter
+            # this check will have to be generalized.
+            return [batch_request]
+
+        # Make a list of the specified and unspecified params in batch_request
+        specified_options = []
+        unspecified_options = []
+        options_template = self.batch_request_options_template()
+        for option_name in options_template.keys():
+            if (
+                option_name in batch_request.options
+                and batch_request.options[option_name] is not None
+            ):
+                specified_options.append(option_name)
+            else:
+                unspecified_options.append(option_name)
+
+        # Make a list of the all possible batch_request.options by expanding out the unspecified
+        # options
+        batch_requests: List[BatchRequest] = []
+
+        if not unspecified_options:
+            batch_requests.append(batch_request)
+        else:
+            # All options are defined by the splitter, so we look at its default values to fill
+            # in the option values.
+            default_option_values = []
+            for option in unspecified_options:
+                default_option_values.append(
+                    self.column_splitter.param_defaults[option]
+                )
+            for option_values in itertools.product(*default_option_values):
```

Comment:
I need to use `itertools` more 😄

Reply:
I was going to implement this but then did some searching. Happy to have found it!